File: 800/python/two.py
t = int(input())

for _ in range(t):
    n, x = map(int, input().split())
    arr = list(map(int, input().split()))
    max_gap = 0
    max_gap = max(max_gap, arr[0])
    for i in range(1, n):
        max_gap = max(max_gap, arr[i] - arr[i-1])
    max_gap = max(max_gap, 2 * (x - arr[-1]))
    
    print(max_gap)

File: 800/two.cpp
#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;

#define rtn return 0;
#define vctr(size) vector<int> arr(size);
#define vctr(n, name) vector<int> name(n); 
#define vinp(n, name) for(int i = 0; i < n; i++) cin >> name[i];
#define FAST_IO ios::sync_with_stdio(false); cin.tie(nullptr);

int main() {
  FAST_IO
  int t;
  cin >> t;
  while(t--){
    int n,x;
    cin>> n >>x;
    vctr(n,arr)
    vinp(n,arr)
    int minimum=0;
    for (size_t i = 0; i <= n; i++)
    {
        if(i==0)
            minimum= max(minimum,arr[i]-0);
        else if(i== n)
            minimum= max(minimum,2*(x-arr[i-1]));
        else
            minimum= max(minimum,arr[i]-arr[i-1]);
    }
    cout<<minimum<<endl;

  }
  
  rtn
}